programs/clearing_house/src/lib.rs

```diff
@@ -929,6 +950,271 @@ pub mod clearing_house {
         Ok(())
     }
 
+    pub fn add_insurance_liquidity<'info>(
+        ctx: Context<AddInsuranceLiquidity>,
+        bank_index: u64,
+        amount: u64,
+    ) -> Result<()> {
+        if amount == 0 {
+            return Err(ErrorCode::InsufficientDeposit.into());
+        }
+
+        let clock = Clock::get()?;
+        // let now = clock.unix_timestamp;
+        let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+
+        let insurance_fund_stake = &mut load_mut!(ctx.accounts.insurance_fund_stake)?;
+        let user_stats = &mut load_mut!(ctx.accounts.user_stats)?;
+
+        validate!(
+            insurance_fund_stake.bank_index == bank_index,
+            ErrorCode::DefaultError,
+            "insurance_fund_stake does not match bank_index"
+        )?;
+
+        validate!(
+            insurance_fund_stake.last_withdraw_request_shares == 0
+                && insurance_fund_stake.last_withdraw_request_value == 0,
+            ErrorCode::DefaultError,
+            "withdraw request in progress"
+        )?;
+
+        let _oracle_map = OracleMap::load(remaining_accounts_iter, clock.slot)?;
+        let bank_map = BankMap::load(&get_writable_banks(bank_index), remaining_accounts_iter)?;
+
+        let _market_map = MarketMap::load(
```

Comment:
probably dont need to load the market/oracles unless we plan to pass the user in here